### PR TITLE
CH4/RMA: Add  RMA window target free preferences

### PR DESCRIPTION
**Pull request by shawnccx**
A new CVAR MPIR_CVAR_CH4_RMA_MEM_EFFICIENT is added to control RMA
window target allocation mode.
If true, memory-saving mode is on.
If false (by default), performance-efficient mode is on.

Fix [issue#2783](https://github.com/pmodels/mpich/issues/2783)

### CODE REVIEW

**Comment by shawnccx** [comment]
I just added this issue number into commit msg.

**Comment by raffenet** [comment]
Thanks, I'll start the tests but leave this up to @minsii to review.
test:jenkins/ch4/most

**Comment by minsii** [comment]
Thanks @raffenet and @shawnccx. I will read it today or tomorrow.

File: src/mpid/ch4/src/ch4r_win.h [hunk 1]
```diff
@@ -20,6 +20,26 @@
 #include <sys/mman.h>
 #endif /* HAVE_SYS_MMAN_H */
 
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_CH4_RMA_MEM_EFFICIENT
+      category    : CH4
+      type        : int
```
[the remark this answers is not in the document]
Comment (minsii) [reply src/mpid/ch4/src/ch4r_win.h]: All others look fine to me.

[the remark this answers is not in the document]
Comment (shawnccx) [reply src/mpid/ch4/src/ch4r_win.h]: Fixed to `boolean` and default to be `false`. Thanks!

**Comment by shawnccx** [comment]
Fixed CVAR type to be `boolean` and set default value to be `false`.

**Pull request merged**